python/paddleserver/paddleserver/model.py

```diff
@@ -58,17 +63,41 @@ def get_model_files(ext: str) -> str:
         self.ready = True
         return self.ready
 
-    def predict(self, payload: Dict, headers: Dict[str, str] = None) -> Dict:
-        instances = payload["instances"]
-        try:
-            inputs = np.array(instances, dtype='float32')
-        except Exception as e:
-            raise Exception("Failed to initialize NumPy array from inputs:%s, %s"
-                            % (e, instances)) from e
-
-        try:
-            self.input_tensor.copy_from_cpu(inputs)
-            self.predictor.run()
-            return {"predictions": self.output_tensor.copy_to_cpu().tolist()}
-        except Exception as e:
-            raise Exception("Failed to predict %s" % e) from e
+    def predict(self, payload: Union[Dict, InferRequest], headers: Dict[str, str] = None) -> Union[Dict, InferResponse]:
+
+        if isinstance(payload, Dict) and "instances" in payload:
+            instances = payload["instances"]
+            try:
+                inputs = np.array(instances, dtype='float32')
+            except Exception as e:
+                raise Exception("Failed to initialize NumPy array from inputs:%s, %s"
```

Comment:
can we use Exception(f"Failed to initialize NumPy array from inputs:{e},{instance}") ? formatting here is kind of funny